### PR TITLE
Real time map

### PR DESCRIPTION
**Pull request by Reshex**
Real time map changes (still not finished)

### CODE REVIEW

File: src/view/pages/statement/components/map/StatementMap.tsx [hunk 1]
```diff
@@ -70,15 +73,42 @@ const StatementMap: FC<Props> = ({ statement }) => {
...
+			try {
+				unsubscribe = await listenToChildStatements(dispatch, statement.statementId, (childStatements) => {
+					setSubStatements(childStatements);
+
+					const topResult = sortStatementsByHirarrchy([
+						statement,
+						...childStatements.filter((state) => isOptionFn(state) || state.statementType === StatementType.question),
+					])[0];
+
+					setResults(topResult);
```
Comment (vladbyk89) [inline src/view/pages/statement/components/map/StatementMap.tsx:97]: Create the callback function outside of the listenToChildStatements props to improve readability.

**Review by vladbyk89 — CHANGES_REQUESTED** [review]
Did you check if your implementation works?
On my local computer it didn't.

**Comment by Reshex** [comment]
> Did you check if your implementation works? On my local computer it didn't.

No it doesn't work properly yet, I've asked for Tal's advice in what should i do in order for it to work. The problem is that when we add a new statement only the new statement is added in real time, the whole map is somehow missing and only on second rendering we get the expected result like before.

**Comment by talyaron** [comment]
Lets talk about it today

**Pull request merged**